sklearn/cluster/k_medoids_.py

```diff
@@ -0,0 +1,292 @@
+        Cluster centers, i.e. medoids (elements from the original dataset)
+
+    labels_ : array, shape = (n_samples,)
+        Labels of each point
+
+    inertia_ : float
+        Sum of distances of samples to their closest cluster center.
+
+    Examples
+    --------
+
+    >>> from sklearn.cluster import KMedoids
+    >>> import numpy as np
+
+    >>> X = np.asarray([[1, 2], [1, 4], [1, 0],
+    ...                 [4, 2], [4, 4], [4, 0]])
+    >>> kmedoids = KMedoids(n_clusters=2, random_state=0).fit(X)
+    >>> kmedoids.labels_
+    array([0, 0, 0, 1, 1, 1])
+    >>> kmedoids.predict([[0,0], [4,4]])
+    array([0, 1])
+    >>> kmedoids.cluster_centers_
+    array([[1, 2],
+           [4, 2]])
+    >>> kmedoids.inertia_
+    8.0
+
+    >>> kmedoids = KMedoids(n_clusters=2, random_state=0,
+    ...                     metric='manhattan').fit(X)
+    >>> kmedoids.labels_
+    array([0, 0, 0, 1, 1, 1])
+    >>> kmedoids.predict([[0,0], [4,4]])
+    array([0, 1])
+    >>> kmedoids.cluster_centers_
+    array([[1, 2],
+           [4, 2]])
+    >>> kmedoids.inertia_
+    8.0
+
+    References
+    ----------
+    Kaufman, L. and Rousseeuw, P.J., Statistical Data Analysis Based on
+    the L1–Norm and Related Methods, edited by Y. Dodge, North-Holland,
+    405–416. 1987
+
+    See also
+    --------
+
+    KMeans
+        The KMeans algorithm minimizes the within-cluster sum-of-squares
+        criterion. It scales well to large number of samples.
+
+
+    Notes
+    -----
+        Since all pairwise distances are calculated and stored in memory for
+        the duration of fit, the space complexity is O(n_samples ** 2).
+    """
+
+    def __init__(self, n_clusters=8, metric='euclidean',
+                 init='heuristic', max_iter=300, random_state=None):
+        self.n_clusters = n_clusters
+        self.metric = metric
+        self.init = init
+        self.max_iter = max_iter
+        self.random_state = random_state
+
+    def _check_nonnegative_int(self, value, desc):
+        """Validates if value is a valid integer > 0"""
+
+        if (value is None or value <= 0 or
+                not isinstance(value, (int, np.integer))):
+            raise ValueError("%s should be a nonnegative integer. "
+                             "%s was given" % (desc, value))
+
+    def _check_init_args(self):
+        """Validates the input arguments. """
+
+        # Check n_clusters and max_iter
+        self._check_nonnegative_int(self.n_clusters, "n_clusters")
+        self._check_nonnegative_int(self.max_iter, "max_iter")
+
+        # Check init
+        init_methods = ['random', 'heuristic']
+        if self.init not in init_methods:
+            raise ValueError("init needs to be one of " +
+                             "the following: " +
+                             "%s" % init_methods)
+
+        # Check random state
+        self.random_state_ = check_random_state(self.random_state)
```

Comment:
This is only used in medoid initialisation. I think you should check and store in a local variable, not an attribute

Reply:
Is this in reference to self.init or self.random_state? I think they're both stored as attributes in KMeans. I think that's the only way they can remain initialization parameters.

Reply:
`self.random_state` is stored by `__init__`, but I don't think the validated `random_state_` needs to be stored.